src/OneTrustCookieConsentBanner.tsx

```diff
@@ -0,0 +1,35 @@
+import { FC, useEffect } from 'react';
+
+type Props = {
+  domainScript: string;
+  enabled: boolean;
+};
+
+const OneTrustCookieConsentBanner: FC<Props> = ({ domainScript, enabled }) => {
+  const src = 'https://cdn.cookielaw.org/scripttemplates/otSDKStub.js';
+
+  const loadOneTrust = () => {
+    const preloadLink = document.createElement('link');
+    preloadLink.rel = 'preload';
+    preloadLink.href = src;
+    preloadLink.as = 'script';
+    document.head.appendChild(preloadLink);
+
+    const script = document.createElement('script');
+    script.src = src;
+    script.type = 'text/javascript';
+    script.dataset.domainScript = domainScript;
+    script.async = true;
+    document.head.appendChild(script);
+  };
+
+  useEffect(() => {
+    if (document && enabled) {
+      loadOneTrust();
+    }
+  }, []);
```

Comment:
dependencies? when enabled changes you need to add it no?

Reply:
true, missed that 👍